src/paho/mqtt/client.py

```diff
@@ -829,7 +829,10 @@ def reconnect(self):
                 ca_certs=self._tls_ca_certs,
                 cert_reqs=self._tls_cert_reqs,
                 ssl_version=self._tls_version,
-                ciphers=self._tls_ciphers)
+                ciphers=self._tls_ciphers,
+                do_handshake_on_connect=False)
```

Comment:
Yes, as it avoid doing the SSL handshake immediately when wrap_socket() is called.

This allow to set a timeout before doing SSL handshake so that the handshake honor the timeout. Without this, with a crappy connection the SSL handshake could hang a long time (and I think I've encounter case where "long time" was days :()